src/control/cmd/dmg/support.go

```diff
@@ -12,13 +12,16 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
+	"time"
 
 	"github.com/daos-stack/daos/src/control/cmd/dmg/pretty"
 	"github.com/daos-stack/daos/src/control/common/cmdutil"
 	"github.com/daos-stack/daos/src/control/lib/control"
 	"github.com/daos-stack/daos/src/control/lib/support"
 )
 
+var bld strings.Builder
```

Comment:
Not sure to understand why you make bld a global variable ?
From what I understand, it should be a variable of the `rsynclog()` function.

Reply:
The requirement is to print all the failure command summary at the end with which node it got failed and the reason. We run multiple command part of multiple function like rsyncLog(), archLogsOnServer() and the main Execute(). If any command failed we need to keep it in the variable for printing at the end.

Reply:
I don't see what we're gaining with the global. Is there some case where the builder isn't being passed along to the helper functions? If not, there should be no difference between this and its old position, as all the usages would be in scope.

Reply:
You mean builder defined in Execute() and pass along to other function? Yes we can do that.

Reply:
Could it not be possible to add it as an attribute of the `collectLogCmd()` class ?
From my side, I would really prefer to avoid global variable.

Reply:
Done